### DISCUSSION TITLE
How to add capillary pressure support for co2 injection case?

### DISCUSSION THREAD

**Discussion by EngSubsurfaceModeling** [GEOS-DEV/GEOS#1743] [General]
Dear all,
````

     I'm trying to use the latest GEOSX develop version to add capillary pressure to an existing case. I tried to add the following,

<BrooksCoreyCapillaryPressure name="capPressure"
                                 phaseNames="{water, gas}"
                                 phaseMinVolumeFraction="{0.01, 0.015}"
                                 phaseCapPressureExponentInv="{0, 6}"
                                 phaseEntryPressure="{0, 1e8}"
                                 capPressureEpsilon="1e-8"/>

and then GEOSX complains that oil phase doesn't exist.  Any suggestions? 
I also noticed that some new syntax in latest GEOSX keywords, I changed some and still fail to make it work. Can anybody help me make it work?
````

[pctest.zip](https://github.com/GEOSX/GEOSX/files/7916265/pctest.zip)

Thanks.

ENG

**Comment by francoishamon** [comment]
Hello @EngSubsurfaceModeling, 

Unfortunately, I may not be able to give a satisfactory answer to your question: 

1- The problem you are observing has been fixed a while ago, and it is now possible to run this case without the oil phase. I attached to this message the input file, and the simulation log. 

[output.txt](https://github.com/GEOSX/GEOSX/files/7917366/output.txt)
[pctest.txt](https://github.com/GEOSX/GEOSX/files/7917367/pctest.txt)

A couple of remarks:

1.a - Don't forget to add `capPressureNames="{ cappres }" ` in the `CompositionalMultiphaseFVM` block
1.b - Don't forget to add `materialList="{ fluid, rock, relperm, cappres }"` in the `CellElementRegion`
1.c - Make sure that the `phaseNames` ordering in `BrooksCoreyCapillaryPressure` is the same as in the other models (i.e., if you use `phaseNames="{ gas, water }"` in `BrooksCoreyRelativePermeability`, GEOSX will output a (meaningful) error message if you put water first in `BrooksCoreyCapillaryPressure`.

With that, the case runs very well.

2- **However**, if you try to run this case with a recent version of GEOSX, you will have the following error in the linear solver: 
`Error!! Block ordering for non-Galerkin coarse grid is not currently supported`. 
I apologize in advance, this comes from the version of the hypre library that we link to at the moment. We hope that it will be resolved soon, but I cannot tell you exactly when.

In the meantime, if you want me to show you a workaround that requires recompiling the TPLs, send me a meeting invite and I will help you do that.
